frontend/src/metabase/visualizations/visualizations/Table.jsx

```diff
@@ -187,7 +199,31 @@ export default class Table extends Component {
           name: col.name,
           fieldRef: col.field_ref,
           enabled: col.visibility_type !== "details-only",
+          key: getColumnKey(col),
         })),
+      getProps: ([
+        {
+          data: { cols },
+        },
+      ]) => ({
+        columns: cols,
+        hasOnEnable: false,
+        listOfSeries: false,
+        extraButton: { text: t`Add or remove columns`, key: "table.columns2" },
+        getItemTitle: columnSetting =>
+          getFriendlyName(
+            findColumnForColumnSetting(cols, columnSetting) || {
+              display_name: "[Unknown]",
+            },
+          ),
+      }),
+    },
+    "table.columns2": {
```

Comment:
Any suggestions for a name for this key?

Reply:
What's this columns2 doing in the application?

Reply:
table.columns2 decides what columns are added and removed. table.columns decides the order. We can't change table.columns though

Reply:
Ideally, it would be great to avoid having duplicated widgets that require adding this workaround `writeSettingId` because it complicates the code. So having only one `"table.columns"` that has two different views for sorting and toggling would be more conventional. But since we are far in this direction, using something that describes the functionality of this settings widget as a key would be nice — `table.columns_visibility` or anything similar

Reply:
The tricky part with 1 widgets with different views was how it interacted with the sidebar (having the back button, changing the title, etc). However, we can improve this in the future. Since that setting def is never written to, we won't be stuck with this if we decide to do something different going forward

Reply:
Maybe a name that is extremely literal.

`table.columnsToBeShownOrHidden` or something of the sort.

Reply:
That would work!